src/ast.c

```diff
@@ -1046,6 +1046,125 @@ int has_meta(jl_array_t *body, jl_sym_t *sym)
     return 0;
 }
 
+void renumber_slots(jl_array_t *code, size_t *newslot);
+
+jl_lambda_info_t *desplat_lambda(jl_lambda_info_t *li, jl_tupletype_t *types)
+{
+    assert(li->isva);
+    size_t *newslot = (size_t*) malloc(jl_array_len(li->slotnames)*sizeof(size_t)+1);
```

Comment:
I think we usually use `alloca` (for small ones) or array/svec (for large ones) for this.